crates/orchestrator/src/data_storage/aws_s3/mod.rs

```diff
@@ -20,11 +22,13 @@ pub struct AWSS3 {
 ///
 /// - initializing a new AWS S3 client
 impl AWSS3 {
-    /// Initializes a new AWS S3 client by passing the config
-    /// and returning it.
-    pub fn new(s3_config: AWSS3Config, aws_config: &SdkConfig) -> Self {
+    /// To init the struct with main settings
+    pub async fn new_with_settings(settings: &impl Settings) -> Self {
+        let s3_config = AWSS3Config::new_with_settings(settings);
+        let aws_config = aws_config::load_from_env().await;
```

Comment:
we are using settings provider and we made the functions assuming that some day some other providers will also be implemented. Idts it is ideal to pass a config of certain provider as an argument to a generalised function. We can discuss it because I don't agree with this design or Ig I am missing something.

Reply:
You're right but what am I saying is that we don't build the aws config by directly reading the env or something. We build it using the settings provider. And then we pass that AWS config downwards, this way 

1. We're still agnostic to what provider is being used
2. We reduce duplication by creating the AWS config only once